pydm/widgets/enum_combo_box.py

```diff
@@ -46,9 +49,22 @@ def set_items(self, enums):
         new_enum_strings : tuple
             The new list of values
         """
+        if not enums:
+            logger.error("Invalid enum value '{0}'. The value is expected to be a valid list of string values."
+                         .format(enums))
+            return
+
         self.clear()
         for enum in enums:
-            self.addItem(enum)
+            if enum is None:
+                logger.error("Invalid enum type '{0}'. The expected type is 'string'.".format(type(enum)))
```

Comment:
if we are enforcing only strings and not doing a cast down there we should be also checking for the `str` type in here, right?
Up for discussion but what if someone wants to point to an array of integer or floats or booleans.... shouldn't we allow and just cast down there... if the cast fails we barf?

What do you say @hmbui, @teddyrendahl and @mattgibbs ?

Reply:
I concur. 

```python
try:
      for enum in enums:
              self.addItem(str(enum))
except Exception as exc:
      logger.exception("Unable to set items %r", enums)
```

Reply:
Unless we have a strict requirement that all elements in the enums tuple must be explicitly str, I don't see a reason to add more type-checking code.

We do have the PyQt API's addItem() taking care of the str conversion, which will raise in case of an invalid value, so let's take advantage of that. That's why we have a catch for TypeError in the proposed changelist.